src/contract.rs

```diff
@@ -53,127 +58,160 @@ pub fn execute(
-        map.insert(metric.addr.clone(), metric);
-    }
-    map
-}
-
-fn query_all_validators_aprs(
+fn query_validators_aprs_by_interval(
     deps: Deps,
     timestamp1: u64,
     timestamp2: u64,
+    from: u32,
+    to: u32,
 ) -> StdResult<Vec<ValidatorAprResponse>> {
-    if timestamp1.eq(&timestamp2) {
+    if timestamp1.ge(&timestamp2) {
         return Err(StdError::GenericErr {
-            msg: "timestamp1 and timestamp2 cannot be the same".to_string(),
+            msg: "timestamp1 cannot be greater than or equal to timestamp2".to_string(),
         });
     }
+    let validators = STATE.load(deps.storage)?.validators;
 
-    let mut response: Vec<ValidatorAprResponse> = vec![];
+    let total_validators: u32 = validators.len().try_into().unwrap();
 
-    let history1_map = convert_validator_metrics_to_map(
-        METRICS_HISTORY.load(deps.storage, U64Key::new(timestamp1))?,
-    );
+    if to.ge(&total_validators) || from > to {
+        return Err(StdError::GenericErr {
+            msg: "Invalid indexes!".to_string(),
+        });
+    }
 
-    let history2 = METRICS_HISTORY.load(deps.storage, U64Key::new(timestamp2))?;
+    let t1 = U64Key::new(timestamp1);
+    let t2 = U64Key::new(timestamp2);
 
-    for h2 in history2 {
-        let h1_opt = history1_map.get(&h2.addr);
-        if let Some(h1) = h1_opt {
-            let apr = compute_apr(h1, &h2, timestamp2 - timestamp1);
+    let mut response: Vec<ValidatorAprResponse> = vec![];
+    let mut start = from;
+
+    while start.le(&to) {
+        let validator_addr = &validators[start as usize];
+        let h1_opt = METRICS_HISTORY.may_load(deps.storage, (&validator_addr, t1.clone()));
+        let h2_opt = METRICS_HISTORY.may_load(deps.storage, (&validator_addr, t2.clone()));
+        if let (Ok(Some(h1)), Ok(Some(h2))) = (h1_opt, h2_opt) {
+            let apr = compute_apr(&h1, &h2, timestamp2 - timestamp1);
             response.push(ValidatorAprResponse { addr: h2.addr, apr });
-        }
+        };
+        start = start + 1;
     }
 
     Ok(response)
 }
 
-fn update_records_to_update_per_run(deps: DepsMut, no: u32) -> Result<Response, ContractError> {
-    STATE.update(deps.storage, |mut s| -> StdResult<_> {
-        s.max_records_to_update_per_run = no;
-        Ok(s)
+fn update_config(
+    deps: DepsMut,
+    info: MessageInfo,
+    batch_size: u32,
+) -> Result<Response, ContractError> {
+    if batch_size == 0 {
+        return Err(ContractError::BatchSizeCannotBeZero {});
+    }
+
+    let manager = CONFIG.load(deps.storage)?.manager;
+    // can only be updated by manager
+    if info.sender != manager {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    CONFIG.update(deps.storage, |mut conf| -> StdResult<_> {
+        conf.batch_size = batch_size;
+        Ok(conf)
     })?;
-    Ok(Response::new().add_attribute("method", "update_records_to_update_per_run"))
+
+    Ok(Response::new().add_attribute("method", "update_config"))
```

Comment:
Add the new batch size. I don't see a point in logging the method here. We can infer that through the ExecuteMsg enum right?